contracts/Faucet.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Faucet
+{
+    // For rate limiting
+    mapping(address => uint256) nextRequestAt;
+    //the ERC20 to be spent
+    IERC20 private _contractAddress;
+    //the owner of the generous token provider for this faucet
+    address private _tokenOwner;
+    //how much tokens we want to spend per claim
+    uint256 private _tokensPerClaim;
+
+    event Claim(address claimer, uint256 amount);
+
+    constructor(address tokenOwner, uint256 tokensPerClaim) {
+        _tokenOwner = tokenOwner;
+        _tokensPerClaim = tokensPerClaim;
+    }
+
+
+    function claim(IERC20 tokenAddress) external {
+        _contractAddress = IERC20(tokenAddress);
```

Comment:
You don't really need the contractAddress variable here. Now the contract always stores the token address of the token, that was last claimed, which is pretty much useless. Just use `IERC20(tokenAddress).transferFrom...`

Reply:
Optionally, you could store an array of "whitelisted" contractAddresses which are valid as input parameters. In this way, you can throw an error, if a user requests tokens from a contract which is not approved.